cmd/registry-replacer/main.go

```diff
@@ -600,6 +635,88 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 	return utilerrors.NewAggregate(errs)
 }
 
+// when pruning unused base images we'll check the fully-resolved config to make sure an image is not used directly in
+// the config, or within any of the tests.
+func pruneUnusedBaseImages(config *api.ReleaseBuildConfiguration, resolvedConfig *api.ReleaseBuildConfiguration) {
+	usedBaseImages := sets.NewString()
+	for _, step := range resolvedConfig.RawSteps {
+		switch {
+		case step.TestStepConfiguration != nil:
+			getTestStepImages(&usedBaseImages, step.TestStepConfiguration)
+		case step.RPMImageInjectionStepConfiguration != nil:
+			usedBaseImages.Insert(string(step.RPMImageInjectionStepConfiguration.From))
+		}
+	}
+	for _, test := range resolvedConfig.Tests {
+		getTestStepImages(&usedBaseImages, &test)
+	}
+	for _, image := range resolvedConfig.Images {
+		usedBaseImages.Insert(string(image.From))
+		for input := range image.Inputs {
+			usedBaseImages.Insert(input)
+		}
+	}
+
+	for sourceImage := range config.BaseImages {
+		var keep bool
+		for _, candidate := range usedBaseImages.List() {
+			orgRepoTag, _ := orgRepoTagFromPullString(candidate)
```

Comment:
Why is the error not being handled?

Reply:
My thought was that it might not be a well-formed pull string and could therefore error out, but we wouldn't care. However, looking more into the `orgRepoTagFromPullString` it should work for all of the valid cases, so if there is an error then it would be a valid error and should be handled.